### PR TITLE
feat(files): grid view 

### LINKED ISSUE #36993: [BITV] 9.2.4.3/3.7 - "Change view" button - The button has been implemented as a checkbox with a label. (2)
Incorrectly, the label element has been made focusable using tabindex="0". Note - the title element should also be removed from label element. 

d8d62a701941522d2687bd0cc6865e6f

<details>
<summary>Details</summary>

https://report.bitvtest.de/default-en/d63601ac-cb34-4645-8256-66bec78964a0.html#checkpoint-89806fe784-v3-n7

</details>

- [ ] 9.2.4.3/10.7 - https://report.bitvtest.de/default-en/d63601ac-cb34-4645-8256-66bec78964a0.html#checkpoint-89806fe784-v10-n7

### PR DESCRIPTION
**Pull request by skjnldsv**
Ref https://github.com/nextcloud/server/issues/39914
Fix https://github.com/nextcloud/server/issues/36993

image

### CODE REVIEW

File: apps/files/src/components/VirtualList.vue [hunk 1]
```diff
@@ -128,11 +134,32 @@ export default Vue.extend({
...
+						key: Object.keys(this.$_recycledPool)[index],
+						item,
+					}
+				}
+
+				// Get and consume reusable key or generate a new one
+				const key = unusedKeys.pop() || Math.random().toString(36).substr(2)
+				this.$_recycledPool[key] = item[this.dataKey]
+				return { key, item }
+			})
```
Comment (skjnldsv) [inline apps/files/src/components/VirtualList.vue:162]: @artonge :star_struck:

File: apps/files/lib/Service/UserConfig.php [hunk 2]
```diff
@@ -47,6 +47,12 @@ class UserConfig {
 			'default' => true,
 			'allowed' => [true, false],
 		],
+		[
+			// Whether to show the files list in grid view or not
+			'key' => 'grid_view',
+			'default' => false,
+			'allowed' => [true, false],
+		],
```
Comment (artonge) [inline apps/files/lib/Service/UserConfig.php:55]: Just a thought: once the new metadata API is done, we could store that per folder, would be fancy, no ?

**Review by artonge — COMMENTED** [review]
As this is mostly moving code around, should we give a closer look to a specific area? Beside the virtual list improvement ;)

File: apps/files/lib/Service/UserConfig.php (hunk 2 above)
Comment (skjnldsv) [reply apps/files/lib/Service/UserConfig.php:55]: There has been some requests. Like https://github.com/nextcloud/server/issues/15170

File: apps/files/src/components/FileEntry/FileEntryBlank.vue [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+  -
+  -->
+<template>
+</template>
+
+<script lang="ts">
+import { translate as t } from '@nextcloud/l10n';
+import Vue from 'vue'
+
+export default Vue.extend({
```
Comment (skjnldsv) [inline apps/files/src/components/FileEntry/FileEntryBlank.vue:29]: This file needs to go before rebase

**Review by artonge — APPROVED** [review]

**Comment by artonge** [comment]
Beware, the base branch seems off

**Comment by skjnldsv** [comment]
> Beware, the target branch seems off

On purpose, please also review https://github.com/nextcloud/server/pull/40854

**Review by szaimen — CHANGES_REQUESTED** [review]
I tested this and in general it seems to work fine :)

But it seems to be broken on mobile currently:
image
image

Also for the first elements the 3-dot menu is not shown:
image

And the list view is currently broken:
image

**Review by szaimen — COMMENTED** [review]
The issues are fixed but the toggle is not shown anymore at all:
image

**Comment by skjnldsv** [comment]
Also done @szaimen

**Comment by skjnldsv** [comment]
PS: not recompiling as I see other PRs are probably going in too

**Review by szaimen — COMMENTED** [review]
Last problem and then ready to go:

on small displays and If in grid view and opening the dropdown on the right it shows multiple scroll bars:
image

**Review by szaimen — APPROVED** [review]
Problem shall be fixed in follow-up

**Pull request merged**